kgforge/specializations/stores/bluebrain_nexus.py

```diff
@@ -351,21 +351,38 @@ def retrieve(
                 url, params=query_params, headers=self.service.headers
             )
             response.raise_for_status()
+        except HTTPError as er:
+            nexus_path = f"{self.service.endpoint}/resources/"
+            if not cross_bucket:
+                nexus_path += f"{self.service.organisation}/{self.service.project}"
+            # Try to use the id as it was given
+            if url_base in id_without_query or nexus_path in id:
+                # Use the given id in the request
+                url = id_without_query
```

Comment:
I would check the retrieve_source so that to add /source:

`if retrieve_source and not cross_bucket:
                    url = "/".join((id_without_query, "source"))
                else:
                    url = id_without_query`

Reply:
In fact, I think the /source does not exist, and the id_without_query should be used directly.

Reply:
What do you mean by `... /source does not exist...` ?
If retrieve_source is true, the requested url should be suffixed with `/source` in order to get the resource payload as the user push it. Actually the same thing as in [L346](https://github.com/BlueBrain/nexus-forge/pull/271/files#diff-9ee1d07daf4890e0d1087d9815baaea67a26d4fd6b31e546784b695165c8e68aR346).

That '/source' missing is the reason why the two resources (retrieved from id and url) are not equal in the section `Retrieving using the resorce url` section of the [`04  - Querying.ipynb`](https://github.com/BlueBrain/nexus-forge/blob/c203996ccf162943e05ba34a0c308e9b9c46180e/examples/notebooks/getting-started/04%20-%20Querying.ipynb) notebook

Reply:
Oh, I think you are correct. I did the change and now it works fine.